### PR TITLE
Add commands related to expenses

### LINKED ISSUE #73: Track expenses
As a student, I want to keep track of my daily, weekly, and monthly expenses so that I can better manage my money.

### LINKED ISSUE #135: [PE-D] some functions about expense are missing
Some functions about expense are missing.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: dmbclub/ped#2

### PR DESCRIPTION
**Pull request by GuoAi**
Add "Spend" command. Still working on other commands related to expenses.

### CODE REVIEW

File: src/main/java/seedu/duke/model/item/Expense.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+public class Expense extends Item {
+
+    private final double value;
+    private String currency;
+    private LocalDate date;
+    private LocalDate now = LocalDate.now();
+
+    /**
+     * Constructor used when adding an expense item.
+     * Byt default, the expense item has currency "SGD and date of today.
```
Comment (iamchenjiajun) [inline src/main/java/seedu/duke/model/item/Expense.java]: Typo in JavaDoc

File: src/main/java/seedu/duke/model/item/Expense.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+ * Represents an expense item.
+ */
+public class Expense extends Item {
+
+    private final double value;
+    private String currency;
+    private LocalDate date;
+    private LocalDate now = LocalDate.now();
```
Comment (iamchenjiajun) [inline src/main/java/seedu/duke/model/item/Expense.java]: Consider creating the variable here and setting it in the constructor instead.

File: src/main/java/seedu/duke/commands/CommandCreator.java [hunk 3]
```diff
@@ -36,13 +37,44 @@ public static Command createAddCommand(String commandString, String description,
...
+        if (argumentsMap.containsKey("currency")) {
+            if (argumentsMap.get("currency").equals("")) {
+                throw new DukeException(Messages.EXCEPTION_EMPTY_CURRENCY);
+            }
+        }
+        if (argumentsMap.containsKey("date")) {
+            if (argumentsMap.get("date").equals("")) {
+                throw new DukeException(Messages.EXCEPTION_EMPTY_DATE);
+            }
+        }
```
Comment (iamchenjiajun) [inline src/main/java/seedu/duke/commands/CommandCreator.java]: Maybe putting these 3 checks in SpendCommand's constructor or execute method would be better, since CommandCreator deals with creating Commands. Checking argumentsMap would be done by the Command itself.

File: src/main/java/seedu/duke/model/itemlist/ItemList.java [hunk 4]
```diff
@@ -53,20 +51,12 @@ public ItemList() {
...
     }
 
-
-    /**
-     * Adds an item into the list.
-     *
-     * @param description the description of the item
-     */
-    public abstract void addTodo(String description);
-
```
Comment (iamchenjiajun) [inline src/main/java/seedu/duke/model/itemlist/ItemList.java]: I think I also changed this in my other PR might have merge conflicts here later. Oops.

Comment (iamchenjiajun) [reply src/main/java/seedu/duke/model/itemlist/ItemList.java]: I renamed mine to `addTaskFromString` and I put it in `TaskList` instead.

File: text-ui-test/EXPECTED.TXT [hunk 5]
```diff
@@ -122,19 +126,19 @@ Hello from...
     ____________________________________________________________
      Got it. I've added this task:
        [T][N] iP meeting (p:0)
-     Now you have 1 task(s) in the list.
+     Now you have 1 items in the list.
```
Comment (iamchenjiajun) [inline text-ui-test/EXPECTED.TXT]: Hmm why is this changed, I thought @MuhammadHoze updated it and this is now the intended behaviour.

File: src/main/java/seedu/duke/model/item/Expense.java (hunk 1 above)
Comment (GuoAi) [reply src/main/java/seedu/duke/model/item/Expense.java]: Thanks for pointing out! Will fix it.

File: src/main/java/seedu/duke/model/item/Expense.java (hunk 2 above)
Comment (GuoAi) [reply src/main/java/seedu/duke/model/item/Expense.java]: Sorry I did not get it. Which variable do you mean?

File: src/main/java/seedu/duke/commands/CommandCreator.java (hunk 3 above)
Comment (GuoAi) [reply src/main/java/seedu/duke/commands/CommandCreator.java]: Sure! Yes it makes sense

File: src/main/java/seedu/duke/model/itemlist/ItemList.java (hunk 4 above)
Comment (GuoAi) [reply src/main/java/seedu/duke/model/itemlist/ItemList.java]: Okay thanks!

File: src/main/java/seedu/duke/model/item/Expense.java (hunk 2 above)
Comment (iamchenjiajun) [reply src/main/java/seedu/duke/model/item/Expense.java]: I was thinking maybe declare the variable `now` in this scope, but put `now = LocalDate.now()` in the constructor. But this seems fine too.

File: text-ui-test/EXPECTED.TXT (hunk 5 above)
Comment (GuoAi) [reply text-ui-test/EXPECTED.TXT]: Oops I did not notice it. I just copied from ACTUAL.TXT to EXPECTED.TXT. I think I might have accidentally changed it when resolving merge conflicts. I will change it back.

**Comment by iamchenjiajun** [comment]
Fixes #135

File: src/main/java/seedu/duke/model/item/Expense.java (hunk 2 above)
Comment (GuoAi) [reply src/main/java/seedu/duke/model/item/Expense.java]: Sure I think assigning values in constructor makes more sense!

**Comment by GuoAi** [comment]
All commands related to expenses are added already. Ready to merge subject to review.

File: expenses.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+f | 0.6 | A | 2020-11-09
```
Comment (MuhammadHoze) [inline expenses.txt:1]: This .txt can be included in the `.gitignore`
